File: python/shaker_sort.py
def main():
    example = [5, 1, 6, 3, 4]
    shaker_sort(example)
    print(example)

def shaker_sort(arr):
    l = len(arr) - 1
    changes = True
    while changes:
        changes = False
        for ind in range(l):
            if arr[ind] > arr[ind + 1]:
                arr[ind], arr[ind + 1] = arr[ind + 1], arr[ind]
                changes = True
        if not changes:
            break
        changes = False
        for ind in range(l, 0, -1):
            if arr[ind] < arr[ind - 1]:
                arr[ind], arr[ind - 1] = arr[ind - 1], arr[ind]
                changes = True
    # no need to return the array

if __name__ == '__main__':
    main()

File: java/shaker_sort.java
public class shaker_sort {
    public static void main(String[] args) {
        int[] example = {5, 1, 6, 3, 4};
        shaker_sort_(example);
        for (int arg : example) {
            System.out.printf("%d ", arg);
        }
    }

    public static void swap(int[] arr, int index1, int index2) {
        int temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
    }

    public static void shaker_sort_(int[] arr) {
        int l = arr.length - 1;
        boolean changes = true;
        while (changes) {
            changes = false;
            for (int ind = 0; ind < l; ind++) {
                if (arr[ind] > arr[ind + 1]) {
                    swap(arr, ind, ind + 1);
                    changes = true;
                }
            }
            if (!changes) break;
            changes = false;
            for (int ind = l; ind > 0; ind--) {
                if (arr[ind] < arr[ind - 1]) {
                    swap(arr, ind, ind - 1);
                    changes = true;
                }
            }
        } // no need to return the array
    }
}
